File: programming/3.3/main.cpp
#include <iostream>
#include <cmath>


using namespace std;

float Yi(int x) {
  return sin(x);
}

int main() {
  float h = 0.1;
  float max = 16;
  int count;
  float sum = 1;
  float r;

  for (float i = 0; i < max; i+=h) {
    r = Yi(i);
    if (r < 0) {
      sum *= r;
      count += 1;
    }
  }

  cout << "Sum:   " << sum << endl;
  cout << "Count: " << count << endl;

  return 0;
}

File: programming/3.3/main.py
from math import sqrt, sin
from random import randint


def Yi(x, h):
  return x + h


def main():
  h = 0.1
  max = 16
  count = 0
  sum = 1
  r = 0
  i = 0

  while i < max:
    i = Yi(i, h)

    if r > 0:
      sum *= r
      count += 1

  print(f'Sum:   {sum}')
  print(f'Count: {count}')


if __name__ == '__main__':
  main()
